hazelcast/src/main/java/com/hazelcast/config/Config.java

```diff
@@ -1394,6 +1396,114 @@ public Config setRingbufferConfigs(Map<String, RingbufferConfig> ringbufferConfi
         return this;
     }
 
+    /**
+     * Returns a read-only AtomicLong configuration for the given name.
+     * <p>
+     * The name is matched by pattern to the configuration and by stripping the
+     * partition ID qualifier from the given {@code name}.
+     * If there is no config found by the name, it will return the configuration
+     * with the name {@code default}.
+     *
+     * @param name name of the AtomicLong config
+     * @return the AtomicLong configuration
+     * @throws ConfigurationException if ambiguous configurations are found
+     * @see StringPartitioningStrategy#getBaseName(java.lang.String)
+     * @see #setConfigPatternMatcher(ConfigPatternMatcher)
+     * @see #getConfigPatternMatcher()
+     */
+    public AtomicLongConfig findAtomicLongConfig(String name) {
+        name = getBaseName(name);
+        AtomicLongConfig config = lookupByPattern(configPatternMatcher, atomicLongConfigs, name);
+        if (config != null) {
+            return config.getAsReadOnly();
+        }
+        return getAtomicLongConfig("default").getAsReadOnly();
+    }
+
+    /**
+     * Returns the AtomicLongConfig for the given name, creating one
+     * if necessary and adding it to the collection of known configurations.
+     * <p>
+     * The configuration is found by matching the the configuration name
+     * pattern to the provided {@code name} without the partition qualifier
+     * (the part of the name after {@code '@'}).
+     * If no configuration matches, it will create one by cloning the
+     * {@code "default"} configuration and add it to the configuration
+     * collection.
+     * <p>
+     * This method is intended to easily and fluently create and add
+     * configurations more specific than the default configuration without
+     * explicitly adding it by invoking {@link #addAtomicLongConfig(AtomicLongConfig)}.
+     * <p>
+     * Because it adds new configurations if they are not already present,
+     * this method is intended to be used before this config is used to
+     * create a hazelcast instance. Afterwards, newly added configurations
+     * may be ignored.
+     *
+     * @param name name of the AtomicLong config
+     * @return the AtomicLong configuration
+     * @throws ConfigurationException if ambiguous configurations are found
+     * @see StringPartitioningStrategy#getBaseName(java.lang.String)
+     * @see #setConfigPatternMatcher(ConfigPatternMatcher)
+     * @see #getConfigPatternMatcher()
+     */
+    public AtomicLongConfig getAtomicLongConfig(String name) {
```

Comment:
We should definitely, at some point in the future, add an solution so all this copy paste logic can be removed.

Reply:
There are a lot of ideas for configuration improvements for HZ 4 (config builder for the user land, immutable data containers for internal usage (no more read-only classes), no more duplication between XSD and `full-example.xml` etc.). There is so much to do on this topic :(